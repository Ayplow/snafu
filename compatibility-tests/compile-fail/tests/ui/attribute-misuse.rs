mod enum_misuse {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(display("display should not work here"))]
    #[snafu(source(from(XXXX, Box::new)))]
    #[snafu(source(true))]
    #[snafu(backtrace)]
    enum EnumError {
        AVariant,
    }
}

mod variant_misuse {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum EnumError {
        // Make sure we catch the error in the second attribute
        #[snafu(display("an error variant"), source(from(XXXX, Box::new)))]
        #[snafu(source)]
        #[snafu(backtrace)]
        AVariant,
    }
}

mod field_misuse {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum EnumError {
        AVariant {
            #[snafu(display("display should not work here"))]
            #[snafu(visibility(pub))]
            #[snafu(source(false))]
            #[snafu(source(from(XXXX, Box::new)))]
            a: String,
        },
    }
}

mod struct_misuse {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum UsableError {}

    #[derive(Debug, Snafu)]
    #[snafu(display("display should not work here"))]
    #[snafu(source(from(UsableError, Box::new)))]
    #[snafu(visibility(pub))]
    #[snafu(source(true))]
    #[snafu(backtrace)]
    struct StructError(Box<UsableError>);
}

fn main() {}
