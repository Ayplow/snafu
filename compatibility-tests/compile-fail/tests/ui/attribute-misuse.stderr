error: `display` attribute is only valid on enum variants, not on an enum
 --> $DIR/attribute-misuse.rs:5:13
  |
5 |     #[snafu(display("display should not work here"))]
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `source(from)` attribute is only valid on enum variant fields, not on an enum
 --> $DIR/attribute-misuse.rs:6:13
  |
6 |     #[snafu(source(from(XXXX, Box::new)))]
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `source(bool)` attribute is only valid on enum variant fields, not on an enum
 --> $DIR/attribute-misuse.rs:7:13
  |
7 |     #[snafu(source(true))]
  |             ^^^^^^^^^^^^

error: `backtrace` attribute is only valid on enum variant fields, not on an enum
 --> $DIR/attribute-misuse.rs:8:13
  |
8 |     #[snafu(backtrace)]
  |             ^^^^^^^^^

error: `context` attribute is only valid on enum variant fields, not on an enum
 --> $DIR/attribute-misuse.rs:9:13
  |
9 |     #[snafu(context)]
  |             ^^^^^^^

error: `source` attribute is only valid on enum variant fields, not on an enum variant
  --> $DIR/attribute-misuse.rs:21:46
   |
21 |         #[snafu(display("an error variant"), source(from(XXXX, Box::new)))]
   |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `source` attribute is only valid on enum variant fields, not on an enum variant
  --> $DIR/attribute-misuse.rs:22:17
   |
22 |         #[snafu(source)]
   |                 ^^^^^^

error: `backtrace` attribute is only valid on enum variant fields, not on an enum variant
  --> $DIR/attribute-misuse.rs:23:17
   |
23 |         #[snafu(backtrace)]
   |                 ^^^^^^^^^

error: `display` attribute is only valid on enum variants, not on a field
  --> $DIR/attribute-misuse.rs:34:21
   |
34 |             #[snafu(display("display should not work here"))]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `visibility` attribute is only valid on an enum and its variants, not on a field
  --> $DIR/attribute-misuse.rs:35:21
   |
35 |             #[snafu(visibility(pub))]
   |                     ^^^^^^^^^^^^^^^

error: Incompatible attributes [`source(false)`, `source(from)`] specified on a field
  --> $DIR/attribute-misuse.rs:37:21
   |
37 |             #[snafu(source(from(XXXX, Box::new)))]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `context` attribute is only valid on enum variant fields, not on a field
  --> $DIR/attribute-misuse.rs:38:21
   |
38 |             #[snafu(context)]
   |                     ^^^^^^^

error: `display` attribute is only valid on enum variants, not on a struct
  --> $DIR/attribute-misuse.rs:51:13
   |
51 |     #[snafu(display("display should not work here"))]
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `visibility` attribute is only valid on an enum and its variants, not on a struct
  --> $DIR/attribute-misuse.rs:53:13
   |
53 |     #[snafu(visibility(pub))]
   |             ^^^^^^^^^^^^^^^

error: `source(bool)` attribute is only valid on enum variant fields, not on a struct
  --> $DIR/attribute-misuse.rs:54:13
   |
54 |     #[snafu(source(true))]
   |             ^^^^^^^^^^^^

error: `backtrace` attribute is only valid on enum variant fields, not on a struct
  --> $DIR/attribute-misuse.rs:55:13
   |
55 |     #[snafu(backtrace)]
   |             ^^^^^^^^^

error: `context` attribute is only valid on enum variant fields, not on a struct
  --> $DIR/attribute-misuse.rs:56:13
   |
56 |     #[snafu(context)]
   |             ^^^^^^^
